server/sonar-server/src/main/java/org/sonar/server/es/IndexerStartupTask.java

```diff
@@ -54,16 +62,55 @@ private boolean indexesAreEnabled() {
   }
 
   private void indexEmptyTypes(StartupIndexer indexer) {
-    Set<IndexType> emptyTypes = getEmptyTypes(indexer);
-    if (!emptyTypes.isEmpty()) {
-      log(indexer, emptyTypes, "...");
-      indexer.indexOnStartup(emptyTypes);
-      log(indexer, emptyTypes, "done");
+    Set<IndexType> uninizializedTypes = getUninitializedTypes(indexer);
+    if (!uninizializedTypes.isEmpty()) {
+      log(indexer, uninizializedTypes, "...");
+      indexer.indexOnStartup(uninizializedTypes);
+      uninizializedTypes.forEach(this::setInitialized);
+      log(indexer, uninizializedTypes, "done");
     }
   }
 
-  private Set<IndexType> getEmptyTypes(StartupIndexer indexer) {
-    return indexer.getIndexTypes().stream().filter(esClient::isEmpty).collect(toSet());
+  private Set<IndexType> getUninitializedTypes(StartupIndexer indexer) {
+    return indexer.getIndexTypes().stream().filter(this::getUninitialized).collect(toSet());
+  }
+
+  private boolean getUninitialized(IndexType indexType) {
+    return esClient.nativeClient().admin().indices().prepareGetSettings(indexType.getIndex()).get().getSetting(indexType.getIndex(),
+      getInitializedName(indexType)) == null;
```

Comment:
If value is manually set to false, then documents should be re-indexed IMO. I suggest to check the value `"true"`.